packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -45,6 +45,11 @@ var homepagePathname = homepagePath ? url.parse(homepagePath).pathname : '/';
 // Webpack uses `publicPath` to determine where the app is being served from.
 // It requires a trailing slash, or the file assets will get an incorrect path.
 var publicPath = ensureSlash(homepagePathname, true);
+// Not everybody wants a single-page app with client side routing without hash.
+// Sometimes, people would rather be able to deploy their app anywhere,
+// without having to specify a "homepage".
+// We can detect this when the publicPath is set to "./".
+var isPublicPathRelative = publicPath === './';
```

Comment:
You're totally right, I did not realize it could be interpreted that way when writing the comment. What do you think of:

```javascript
// Some apps built with Create React App do not use client-side routing with pushState.
// For these, "homepage" can be set to "./" to make them deployable anywhere.
```